application/models/team.py

```diff
@@ -0,0 +1,63 @@
+from application.errors import NotFound, NotInitialized
+from flask import current_app
+from py2neo import Graph, Node
+import uuid
+
+
+class Team:
+
+    @classmethod
+    def all(cls):
+        results = current_app.graph.run("MATCH (x:Team) RETURN x")
+        teams = []
+        for result in results:
+            team = Team(result['x']['name'])
+            team.id = result['x']['id']
+            teams.append(team)
+        return teams
+
+    @classmethod
+    def find_by(cls, force=False, id='', name=''):
```

Comment:
Hardcoding the search parameters in the arguments does not scale very well.
Maybe you can pass a hash of search parameters and append an `OR x.{key} = '{value}'` to the query for each member of the hash.

Reply:
You have to adjust the Cypher query individually for all search params anyways, and I don't think we will add THAT many fields to the Team Node. I agree that a hash would maybe be nice but then the function call also looks somewhat messy with `Team.find_by({"id": some_uuid})` instead of `Team.find_by(id=some_uuid)`

I'll have a look at how django does it maybe.

Reply:
I like the params hash more too, alternatively you could use `find_by(**kwargs)` for passing an arbitrary number of named parameters.

Reply:
yes **kwargs should be used for this but not right now as there are only 2 params to use. Will be refactored once we add more to the model.